code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -4,9 +4,12 @@
 
 	if(def_zone)
 		if(isbodypart(def_zone))
-			return checkarmor(def_zone, type)
+			var/obj/item/bodypart/bp = def_zone
+			if(bp.body_part)
+				return checkarmor(def_zone, type)
 		var/obj/item/bodypart/affecting = get_bodypart(ran_zone(def_zone))
-		return checkarmor(affecting, type)
+		if(affecting.body_part)

```

Comment:
this will runtime if affecting is null

Reply:
get_bodypart returns bodyparts that are in a mob's body_parts list, when a body part is dismembered, it is also removed from the list so i dont think it'll happen

Reply:
but you're checking a var of something that could potentially be null here, no?

Reply:
Yes, but say they're missing a leg. ran_zone doesn't check if the zone exists, and can return said leg to get_bodypart, which will then return null
now you're checking a var of something that's null, which is what's causing the runtime in the first place.

Reply:
oop im rarted